test_log/src/main/java/org/corfudb/test_log/TestLog.java

```diff
@@ -0,0 +1,289 @@
+package org.corfudb.test_log;
+
+import org.apache.commons.codec.binary.Hex;
+import org.corfudb.format.Types.LogEntry;
+import org.corfudb.format.Types.LogHeader;
+import org.corfudb.format.Types.Metadata;
+import org.corfudb.format.Types.DataType;
+
+import java.nio.ByteBuffer;
+import java.lang.String;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.nio.channels.FileChannel;
+import java.io.IOException;
+import java.util.Map;
+
+import com.google.protobuf.ByteString;
+import com.google.common.hash.Hashing;
+import com.google.common.hash.HashFunction;
+import com.google.common.hash.HashCode;
+import org.docopt.Docopt;
+
+public class TestLog {
+    private static int MetadataSize;
+    private static int LogHeaderSize;
+    private static final String USAGE =
+            "Usage:\n"
+                    + "\ttest_log [-d <show binary> | -e <address> | -E <address>]"
+                    + "\n"
+                    + "Options:\n"
+                    + " -d <show binary>, --display=<show binary>   Display log records (include binary data)\n"
+                    + " -e <address>, --erase=<address>             Erase log record\n"
+                    + " -E <address>, --erase_remain=<address>      Erase log records to end of file\n";
+
+    public static void main(String[] args) {
+        Map<String, Object> opts = new Docopt(USAGE).parse(args);
+        Operation op = new Operation(Operation.OperationType.REPORT, 0);
+        boolean use_output_file = false;
+        if (opts.get("--erase") != null) {
+            int addr = Integer.parseInt((String) opts.get("--erase"));
+            op = new Operation(Operation.OperationType.ERASE_ONE, addr);
+            use_output_file = true;
+        } else if (opts.get("--erase_remain") != null) {
+            int addr = Integer.parseInt((String) opts.get("--erase_remain"));
+            op = new Operation(Operation.OperationType.ERASE_REMAINING, addr);
+            use_output_file = true;
+        } else if (opts.get("--display") != null) {
+            int flag = Integer.parseInt((String) opts.get("--display"));
+            if (flag == 0) {
+                op = new Operation(Operation.OperationType.DISPLAY, 0);
+            } else {
+                op = new Operation(Operation.OperationType.DISPLAY_ALL, 0);
+            }
+        }
+        LogHeader lh = LogHeader.newBuilder()
+                .setVersion(10)
```

Comment:
Code is chopped up in this format but when I look at the whole file I think this instance is a dummy that is constructed to capture the size of the instance.